Moved some functions to base image

PriyaBasker23 opened the pull request:
Moved some common functions from resync processed file  container to base image

PriyaBasker23 committed:
Moved some functions to base image

PriyaBasker23 committed:
Added change log

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -108,6 +108,25 @@ def get_account_id() -> str:
+        raise ValueError(f"{regex} not found in {string}")
+
+
+def database_name_regex() -> str:
```
Rather than exposing the regex itself, could we have functions for each one, like 

- `extract_timestamp_from_curated_path(string: str)`
- `extract_table_name_from_curated_path(string: str)`
- `extract_database_name_from_curated_path(string: str)`

it should simplify the calling code

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -108,6 +108,25 @@ def get_account_id() -> str:
+
+
+def extraction_timestamp_regex() -> str:
+    return """(extraction_timestamp=[^\/]*)\/"""  # noqa: W605
```
There is no need to surpress W605 here, this warning is due to python trying to interpret `\/` as an escape sequence before it gets to the regex library.

For regexes you should use raw strings like `r"(extraction_timestamp=[^\/]*)\/"`

MatMoore replied:
This particular one already exists as a precompiled regex at the start of the module, 

```python
EXTRACTION_TIMESTAMP_REGEX = re.compile(
    r"^(.*)/(extraction_timestamp=)([0-9TZ]{1,16})/(.*)$"
)
```

I think this is doing much the same thing except it's anchored at the end of the string - so we should choose one regex or the other.

It's unlikely to make a significant performance difference at the moment, but in general precompiling the regex with re.compile will speed things up if the regex gets called a lot of times (e.g. in deeply nested loops).

PriyaBasker23 committed:
refactored  functions

PriyaBasker23 committed:
code refactor

PriyaBasker23 committed:
Fixed lint error

PriyaBasker23 committed:
fixed lint error

PriyaBasker23 committed:
fixed uper linter error

PriyaBasker23 committed:
fixed lint error

PriyaBasker23 committed:
Fixed unit test

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -27,8 +27,10 @@
 EXTRACTION_TIMESTAMP_REGEX = re.compile(
     r"^(.*)/(extraction_timestamp=)([0-9TZ]{1,16})/(.*)$"
 )
+DATABASE_NAME = re.compile(r"database_name=([^\/]*)\/")
```
```suggestion
DATABASE_NAME_REGEX = re.compile(r"database_name=([^\/]*)\/")
```

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -27,8 +27,10 @@
     r"^(.*)/(extraction_timestamp=)([0-9TZ]{1,16})/(.*)$"
 )
+DATABASE_NAME = re.compile(r"database_name=([^\/]*)\/")
+TABLE_NAME = re.compile(r"table_name=([^\/]*)\/")
```
```suggestion
TABLE_NAME_REGEX = re.compile(r"table_name=([^\/]*)\/")
```

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -109,6 +111,29 @@ def get_account_id() -> str:
+
+
+def extract_table_name_from_curated_path(string: str):
+    return search_string_for_regex(string, regex=TABLE_NAME)
```
```suggestion
    return search_string_for_regex(string, regex=TABLE_NAME_REGEX)
```

MatMoore commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -109,6 +111,29 @@ def get_account_id() -> str:
+
+
+def extract_database_name_from_curated_path(string: str):
+    return search_string_for_regex(string, regex=DATABASE_NAME)
```
```suggestion
    return search_string_for_regex(string, regex=DATABASE_NAME_REGEX)
```

MatMoore reviewed (dismissed):
Looks good, just a minor naming suggestion, take it or leave it 🙂

PriyaBasker23 committed:
updated some name changes

PriyaBasker23 committed:
Fixed lint error

PriyaBasker23 committed:
Added return statement

PriyaBasker23 committed:
added return typing

PriyaBasker23 committed:
Changed the return type

PriyaBasker23 committed:
Changed the return type

tom-webber commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -109,6 +114,26 @@ def get_account_id() -> str:
+def search_string_for_regex(string: str, regex: re.Pattern[str]) -> Optional[str]:
+    """Search a string for a regex pattern and return the first result"""
+    search_output = regex.search(string)
+    return search_output.groups()[0] if search_output else None
```
```suggestion
def search_string_for_regex(string: str, regex: re.Pattern[str]) -> str:
    """Search a string for a regex pattern and return the first result"""
    search_match = regex.search(string)
    if not search_match:
        raise Exception((f"{regex} not found in {string}"))
    return search_match.group(0)
```
Added an error catching for if the regex isn't matched, as `None` isn't handled by downstream uses.
[re.search only matches the first occurrence of the string](https://docs.python.org/3/library/re.html#functions), returning a match with only that group.

PriyaBasker23 replied:
For my uses case,  if directory structure does match returning None will be useful hence i didnt want to Raise exception

tom-webber replied:
I changed it here because None isn't handled in downstream use cases. 

If you want to return None, I'd recommend logging `f"{regex} not found in {string}"` instead of `raise`, then handle `None` in the downstream method calls that at the moment are expecting a string.

PriyaBasker23 replied:
ok

tom-webber commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -19,6 +19,7 @@
 from datetime import datetime
 from typing import NamedTuple
 from uuid import UUID, uuid4
+from typing import Optional
```
```suggestion
from typing import NamedTuple, Optional
from uuid import UUID, uuid4
```
linting

PriyaBasker23 replied:
Thanks

tom-webber replied:
FYI You can hit 'Commit suggestion' to add the change

PriyaBasker23 committed:
Fixed lint error

PriyaBasker23 committed:
Fixed lint error

PriyaBasker23 committed:
Fixed black error

PriyaBasker23 committed:
sorted black error

PriyaBasker23 committed:
isort fixes

PriyaBasker23 committed:
fixing isort issues

PriyaBasker23 committed:
fixing isort issues

Commit:
Add pyproject.toml to specify black as isort profile
superlinter uses above for isort

Commit:
fix workspace path ref

Commit:
as above

Commit:
log message if regex not found (rather than return)
add logging rather than printing (cannot use custom logger as it would create circular dependency)
black import linting

tom-webber commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -27,8 +28,12 @@
+TABLE_NAME_REGEX = re.compile(r"table_name=([^\/]*)\/")
+
 
-# https://docs.aws.amazon.com/athena/latest/ug/tables-databases-columns-names.html
```
This feels like a useful comment for context. I would leave it in.

tom-webber reviewed (dismissed):
lgtm

PriyaBasker23 committed:
added comments
